sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -91,6 +91,16 @@
 # GcsIO.delete_batch().
 MAX_BATCH_OPERATION_SIZE = 100
 
+# Batch endpoint URL for GCS.
+# We have to specify an API specific endpoint here since Google APIs global
+# batch endpoints will be deprecated on 03/25/2019.
+# See https://developers.googleblog.com/2018/03/discontinuing-support-for-json-rpc-and.html.
+# Currently apitools library uses a global batch endpoint by default: https://github.com/google/apitools/blob/master/apitools/base/py/batch.py#L152
+# TODO: remove this constant and it's usage after apitools move to using an API
```

Comment:
Let's create a JIRA to move off of apitools. That library itself is in maintenance mode.

Reply:
Created https://issues.apache.org/jira/browse/BEAM-4055